examples/energy-management-app/linux/main.cpp

```diff
@@ -35,80 +36,181 @@ using namespace chip;
 using namespace chip::app;
 using namespace chip::app::Clusters;
 
-static EnergyEvseDelegate * gDelegate       = nullptr;
-static EnergyEvseManager * gInstance        = nullptr;
-static EVSEManufacturer * gEvseManufacturer = nullptr;
+static EnergyEvseDelegate * gEvseDelegate            = nullptr;
+static EnergyEvseManager * gEvseInstance             = nullptr;
+static DeviceEnergyManagementDelegate * gDEMDelegate = nullptr;
+static DeviceEnergyManagementManager * gDEMInstance  = nullptr;
+static EVSEManufacturer * gEvseManufacturer          = nullptr;
```

Comment:
I think these should be std::unique_ptr to have less of a burden on new/delete ordering (and make leak checkers work better if we ever run them). Unless there is a dependency order on delete, in which case we should document those.

Reply:
Changed to use unique_ptr here. There is some ordering which is now documented.